### PR TITLE
Security Fix - Solution for Issue #1389

### PR DESCRIPTION
**Pull request by priya-anand**
For this XSS - if we could sanitize inputs in these two lines of codes, it would take care of the group admins - I prefer this method as it needs minimal amount of changes to address this issue.

### CODE REVIEW

File: gacl/admin/edit_group.php [hunk 1]
```diff
@@ -84,12 +84,12 @@
 		//No parent, assume a "root" group, generate a new parent id.
 		if (empty($_POST['group_id'])) {
 			$gacl_api->debug_text('Insert');
-			
-			$insert_id = $gacl_api->add_group($_POST['value'], $_POST['name'], $parent_id, $group_type);
+			// fix for XSS vulnerability 
+			$insert_id = $gacl_api->add_group(htmlspecialchars($_POST['value']), htmlspecialchars($_POST['name']), $parent_id, $group_type);
```
Comment (bradymiller) [inline gacl/admin/edit_group.php]: In this case, are you placing html escaped code in the database?

File: gacl/admin/edit_group.php [hunk 2]
```diff
@@ -84,12 +84,12 @@
...
 		} else {
 			$gacl_api->debug_text('Update');
-			
-			$gacl_api->edit_group($_POST['group_id'], $_POST['value'], $_POST['name'], $parent_id, $group_type);
+			// fix for XSS vulnerability 
+			$gacl_api->edit_group($_POST['group_id'], htmlspecialchars($_POST['value']), htmlspecialchars($_POST['name']), $parent_id, $group_type);
```
Comment (bradymiller) [inline gacl/admin/edit_group.php]: In this case, are you placing html escaped code in the database?

**Comment by bradymiller** [comment]
hi @priya-anand ,
It looks like this is inserting html escaped code into the database, which we should avoid (data is not always sent to html).
Check out the following PR for an example of html escaping this script:
https://github.com/openemr/openemr/pull/1404
(note it deals with the really odd mechanism gacl uses to make the spacing with an img)

**Comment by priya-anand** [comment]
I agree - this would push the escaped code into the database - not a good option.

**Pull request closed without merging**